kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -54,12 +55,6 @@ class CDCL {
      */
     val watchers: MutableList<MutableList<Clause>> = mutableListOf()
 
-    /**
-     * The number of variables.
-     */
-    var numberOfVariables: Int = 0
```

Comment:
This field is a part of public API

Reply:
Number of variables is complicated now because there are two counters for that: number of variables from the CNF, and number of non-substituted variables actually used in the problem. It will be worse after BVE. I can leave a public getter to `assignment.numberOfVariables` if needed, but I won't for now. We will cross that bridge once we get to it.

Reply:
Yea, this is also a "problem" in Minisat and Cadical: that `numberOfVariables` does not represent the number of variables in the original CNF. We either should make a distinction between "active" (decision-able) and "eliminated" variables, or just store the largest variable index. It depends on how this field (`numVars`) is going to be used by users. Personally, I would be happy with the second (just get the largest used variable index from the solver, in order to be able to rely on the "weak invariant" `newVar() == numVars+1`). In addition, if I ever need "number of active/eliminated variables", I would expect separate fields/getters to exist for this matter.